File: src/ru/gurps/generator/models/Language.java
package ru.gurps.generator.models;

import ru.gurps.generator.Main;
import ru.gurps.generator.config.Model;

public class Language extends Model {
    public Integer id;
    public String name;
    @Ignore public Integer spoken = 0;
    @Ignore public Integer written = 0;
    @Ignore public Integer cost = 0;
    @Ignore public Boolean add = false;

    public Language() {
    }

    public Language(String language) {
        this.name = language;
    }

    public String getName() {
        return name;
    }

    public String getSpoken() {
        switch(spoken){
            case 0: return Main.locale.getString("not_have");
            case 1: return Main.locale.getString("broken");
            case 2: return Main.locale.getString("accent");
            case 3: return Main.locale.getString("native");
        }
        return null;
    }

    public void setSpoken(String spoken) {
        if(spoken.equals(Main.locale.getString("not_have"))) this.spoken = 0;
        else if(spoken.equals(Main.locale.getString("broken"))) this.spoken = 1;
        else if(spoken.equals(Main.locale.getString("accent"))) this.spoken = 2;
        else if(spoken.equals(Main.locale.getString("native"))) this.spoken = 3;
    }

    public String getWritten() {
        switch(written){
            case 0: return Main.locale.getString("illiteracy");
            case 1: return Main.locale.getString("semi-literate");
            case 2: return Main.locale.getString("literacy");
        }
        return null;
    }

    public void setWritten(String written) {
        if(written.equals(Main.locale.getString("illiteracy"))) this.written = 0;
        else if(written.equals(Main.locale.getString("semi-literate"))) this.written = 1;
        else if(written.equals(Main.locale.getString("literacy"))) this.written = 2;
    }

    public String getCost() {
        return Integer.toString(cost);
    }

    public Boolean getAdd() {
        return add;
    }
}

File: src/main/kotlin/ru/gurps/generator/desktop/models/rules/Language.kt
package ru.gurps.generator.desktop.models.rules

import ru.gurps.generator.desktop.config.Model
import ru.gurps.generator.desktop.singletons.LanguagesValues
import tornadofx.FX
import tornadofx.get
import java.sql.ResultSet

class Language : Model {
    override var id: Int = -1
    var name: String = ""
    var spoken: Int = 0
    var written: Int = 0
    var cost: Int = 0
    var add: Boolean = false

    constructor()
    constructor(result: ResultSet) {
        id = result.getInt("id")
        val nameRes = result.getString("name")
        if (nameRes != null) name = nameRes
    }

    constructor(language: String) {
        this.name = language
    }

    fun getCost(): String {
        return cost.toString()
    }

    fun getSpoken(): String {
        return LanguagesValues.spoken[spoken]
    }

    fun setSpoken(spoken: String) {
        when(spoken) {
            FX.messages["not_have"] -> this.spoken = 0
            FX.messages["broken"] -> this.spoken = 1
            FX.messages["accent"] -> this.spoken = 2
            FX.messages["native"] -> this.spoken = 3
        }
    }

    fun getWritten(): String? {
        return LanguagesValues.spoken[written]
    }

    fun setWritten(written: String) {
        when (written) {
            FX.messages["illiteracy"] -> this.written = 0
            FX.messages["semi-literate"] -> this.written = 1
            FX.messages["literacy"] -> this.written = 2
        }
    }
}
